src/server/lib/locale-handler.js

```diff
@@ -0,0 +1,18 @@
+import * as cookie from '../lib/cookie'
+
+/**
+ * Get locale based on body / query param / cookie
+ */
+export default function localeHandler (req, res, cookieOptions) {
+  const { body, query } = req
+  // get locale from body, query or cookie
+  const localeCookieValue = req.cookies[cookieOptions.locale.name] || null
```

Comment:
Seeing as the cookie in question changes routing behavior and is actually supposed to be set by language switching logic on the website itself, I don't think next-auth should set a value to this cookie as it may cause behavior changes the user may not expect. We could use it as one of the fallbacks if it is defined however?